app/main.py

```diff
@@ -1,64 +1,15 @@
 """
-FastAPI Test
+Hauptmodul der App.
 """
+
 from fastapi import FastAPI
-from pydantic import BaseModel
+from app.routers import clustering_router
 
 app = FastAPI()
 
+app.include_router(clustering_router.router, prefix="/clustering", tags=["clustering"])
 
-class Item(BaseModel):
-    """
-        Item Klasse
-    """
-    id: int
-    name: str
-    description: str | None = None
-    price: float
-    tax: float | None = None
-    tags: list[str] = []
-
-
-@app.get("/hello")
-def hello_world():
-    """
-    Hello World
-
-    :return:
-    """
-    return {"hello": "test deploy"}
-
-
-@app.post("/items/")
-def create_item(item: Item) -> Item:
-    """
-    Create Item Test
-
-    :param item:
-    :return:
-    """
-    return item
-
-
-@app.get("/items/")
-def read_items() -> list[Item]:
-    """
-    List Item Test
-
-    :return:
-    """
-    return [
-        Item(name="Portal Gun", price=42.0, id=1),
-        Item(name="Plumbus", price=32.0, id=2),
-    ]
-
-
-@app.get("/items/{id}")
-def read_item(item_id: int) -> Item:
-    """
-    List One Item Test
-
-    :param item_id:
-    :return:
-    """
-    return Item(name="Portal Gun", price=42.0, id=item_id)
+# Optional: Wenn die App direkt ausgeführt werden soll
+# if __name__ == "__main__":
+#     import uvicorn
+#     uvicorn.run(app, host="0.0.0.0", port=8000)
```

Comment:
Die App soll direkt ausgeführt werden wenn der docker startet...

Reply:
Das ist hier nicht notwendig und kann eigentlich gelöscht werden. Das liegt daran, dass im Dockerfile bereits angegeben wird, dass Uvicorn direkt beim Start des Containers die App mit den angegebenen Host- und Port-Optionen ausführen soll.

In diesem speziellen Fall macht der CMD-Befehl im Dockerfile:

CMD ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8080"]

genau das -> app starten

Reply:
Stimmt. Nicht dran gedacht, sorry :) Rein in die Masse